src/gc/gc.d

```diff
@@ -2803,19 +2799,36 @@ struct Pool
         }
     }
 
-    void clrBitsSmallSweep(size_t dataIndex, GCBits.wordtype toClear) nothrow
+    void freePageBits(size_t pagenum, in ref PAGEBITS toFree) nothrow
     {
-        immutable toKeep = ~toClear;
+        import core.internal.traits : staticIota;
+
+        immutable beg = pagenum * (PAGESIZE / 16 / GCBits.BITS_PER_WORD);
+
+        foreach (i; staticIota!(0, PAGEBITS.length))
```

Comment:
Have you tried making this a single loop that can be shortcut for `toFree[i] == 0`?